### PR TITLE
preliminary refactoring for the Multi-Threading support

### PR DESCRIPTION
**Pull request by artpol84**
**Status**: 
* PR seems to be ready to go (passing a simple test at least).
* The PR is based on #11 and will have to be rebased against master once/if #11 is merged.

### CODE REVIEW

File: report.c [hunk 1]
```diff
@@ -1953,91 +1926,75 @@ mpiPi_coll_print_all_callsite_time_info (FILE * fp)
...
-	  task_lookup->cumulativeTime = 0;
-	  task_lookup->cumulativeTimeSquared = 0;
-	  task_lookup->maxDur = 0;
-	  task_lookup->minDur = 0;
-	  task_lookup->cumulativeDataSent = 0;
-	  task_lookup->cumulativeIO = 0;
-	  task_lookup->maxDataSent = 0;
-	  task_lookup->minDataSent = 0;
-	  task_lookup->maxIO = 0;
-	  task_lookup->minIO = 0;
```
Comment (artpol84) [inline report.c]: @cchambreau @vetter Should it be 
```C
task_lookup->minIO = DBL_MAX
```

Comment (cchambreau) [reply report.c]: Yes, thanks!

File: mpiPconfig.h.in [hunk 2]
```diff
@@ -15,6 +15,10 @@
 #ifndef _MPIPCONFIG_H
 #define _MPIPCONFIG_H
 
+#define MPIP_CALLSITE_STACK_DEPTH_MAX 8
```
Comment (artpol84) [inline mpiPconfig.h.in]:
```suggestion
#define MPIP_CALLSITE_STACK_DEPTH_MAX @STACKDEPTH@	
```

File: mpiPconfig.h.in [hunk 3]
```diff
@@ -15,6 +15,10 @@
...
+#define MPIP_HELP_LIST "mpip-help@googlegroups.com"
```
Comment (artpol84) [inline mpiPconfig.h.in]:
```suggestion
 #define MPIP_HELP_LIST "@PACKAGE_BUGREPORT@"
```

**Comment by artpol84** [comment]
> Getting rid of mpiPi.h.in looks good, but could we get the STACKDEPTH and PACKAGE_BUGREPORT variables into mpiPconfig.h.in?

Good catch! Fixed

**Comment by artpol84** [comment]
>> Getting rid of mpiPi.h.in looks good, but could we get the STACKDEPTH and PACKAGE_BUGREPORT variables into mpiPconfig.h.in?

> Good catch! Fixed

@cchambreau actually, it is not fixed.
I missed that you have autoconf in place. I will make a proper fix and incorporate in the right commit.
What confused me is the presense of configure, but thats totally my bad.

**Comment by artpol84** [comment]
I suggest to merge #16 first and I'll rebase on top of it.

**Comment by artpol84** [comment]
This PR is now rebased on top of #16 and is using updated configury logic.

**Pull request merged**